### PR TITLE
Enable location and debug info.

### PR DESCRIPTION
**Pull request by maleadt**
We'll probably have to disable this on older drivers, but let's see what CI thinks first.

Requires https://github.com/JuliaGPU/GPUCompiler.jl/pull/182.

### CODE REVIEW

**Comment by maleadt** [comment]
Still seems to segfault on driver 465 :disappointed: Filed with NVIDIA as bug #3305774.

**Comment by maleadt** [comment]
Interestingly, this doesn't happen using `ptxas` from the CUDA 11.3 toolkit... So that's what prompted https://github.com/JuliaGPU/CUDA.jl/pull/892.
EDIT: so that's false, it still reproduces using `ptxas` but requires `-g`.

**Comment by maleadt** [comment]
I think we may be able to re-enable this: it looks like lineinfo (as emitted on `-g1`) doesn't trip up `ptxas` anymore, and while debug info (from `-g2`) does, that's not enabled by default and users can opt out by not running on `-g2` or by setting `JULIA_CUDA_DEBUG_INFO` to `false`. But that's a temporary thing, I reduced all crashes I encountered to the same PTX pattern which NVIDIA has mentioned should be fixed on a future version of CUDA.

**Comment by maleadt** [comment]
Should probably wait for https://github.com/JuliaGPU/CUDA.jl/issues/955 before merging this.

File: src/compiler/execution.jl [hunk 1]
```diff
@@ -312,6 +312,18 @@ end
...
 
+    # remove extraneous debug info on lower debug levels
+    if Base.JLOptions().debug_level < 2
+        # even though we won't actually pass `--device-debug` to ptxas, `.target debug` also
+        # seems to enable debug-mode compilation, which triggers NVIDIA bug #3305774
+        #
+        # it would be better if GPUCompiler.jl could configure LLVM to only emit location
+        # tracking instructions without actual DWARF info. this used to be possible using
+        # LocTrackingOnly/EmitDebugInfo from D4234, but got removed in favor of NoDebug in
+        # D18808, seemingly breaking the use case of only emitting `.loc` instructions...
```
Comment (vchuravy) [inline src/compiler/execution.jl]: So `LLVMDebugEmissionKindLineTablesOnly` is not sufficient?

Comment (maleadt) [reply src/compiler/execution.jl]: Correct. It results in a `.target debug` entry in the PTX, which essentially triggers full device debug mode.

**Pull request merged**